xtask/src/utils.rs

```diff
@@ -1,47 +1,15 @@
 use anyhow::ensure;
 use std::process::Command;
 
-// Replacement for str.split_once() on Rust older than 1.52
-#[rustversion::before(1.52)]
-pub fn split_once(s: &str, pat: char) -> Option<(&str, &str)> {
-    let mut iter = s.splitn(2, pat);
-    Some((iter.next()?, iter.next()?))
-}
-
-#[rustversion::since(1.52)]
-pub fn split_once(s: &str, pat: char) -> Option<(&str, &str)> {
-    s.split_once(pat)
-}
-
-#[rustversion::since(1.57)]
-pub fn format_command(command: &Command) -> String {
-    let mut buf = String::new();
-    buf.push('`');
-    buf.push_str(&command.get_program().to_string_lossy());
-    for arg in command.get_args() {
-        buf.push(' ');
-        buf.push_str(&arg.to_string_lossy());
-    }
-    buf.push('`');
-    buf
-}
-
-#[rustversion::before(1.57)]
-pub fn format_command(command: &Command) -> String {
-    // Debug impl isn't as nice as the above, but will do on < 1.57
```

Comment:
I chose to drop the distinction to get rid of rustversion entirely. I think this is possible since we do not use xtask in the CI any more and any one using it for local development will most likely be running stable or nightly.

At the danger of opening old wounds, I would like to bring up again that we currently have two xtask-like mechanisms and I think the emphasis has definitely shifted towards Nox. Having another one that is rarely exercised seems confusing to contributors at least. I also wonder whether we want to invest the effort for modernising this, e.g. replacing struct and clap v2 by clap v4 wholesale. Personally, I would suggest dropping the Rust-based build automation.